torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -405,7 +445,14 @@ void TensorPipeAgent::pipeWrite(
     std::function<void(const tensorpipe::Error&)> fn) {
   tensorpipe::Message tpMessage;
   TensorpipeWriteBuffers tpBuffers;
-  std::tie(tpMessage, tpBuffers) = tensorpipeSerialize(std::move(rpcMessage));
+
+  const auto& deviceMaps =
+      rpcMessage.isRequest() ? opts_.deviceMaps : reverseDeviceMaps_;
+  std::tie(tpMessage, tpBuffers) = tensorpipeSerialize(
+      std::move(rpcMessage),
+      getDevicesForTensors(
```

Comment:
@lw I noticed that this `TORCH_CHECK` failed in this function, the following line takes a lot longer to finish (probably waiting for timeout). Does it mean we should not crash in `pipeWrite`, and instead should detect the error in the call site?


https://github.com/pytorch/pytorch/blob/0134deda0fb149a8088b15c409617b98c89b34c1/torch/csrc/distributed/rpc/tensorpipe_agent.cpp#L808